redis/connection.py

```diff
@@ -734,6 +715,188 @@ def _host_error(self):
         return f"{self.host}:{self.port}"
 
 
+def ensure_string(key):
+    if isinstance(key, bytes):
+        return key.decode("utf-8")
+    elif isinstance(key, str):
+        return key
+    else:
+        raise TypeError("Key must be either a string or bytes")
+
+
+class CacheProxyConnection(ConnectionInterface):
+    def __init__(
+        self,
+        conn: ConnectionInterface,
+        cache: Cache,
+        conf: CacheConfiguration,
+        cache_lock: threading.Lock,
+    ):
+        self.pid = os.getpid()
+        self._conn = conn
+        self.retry = self._conn.retry
+        self.host = self._conn.host
+        self.port = self._conn.port
+        self._cache = cache
+        self._cache_lock = cache_lock
+        self._conf = conf
+        self._current_command_hash = None
+        self._current_command_keys = None
+        self._current_options = None
+        self._keys_mapping = LRUCache(maxsize=10000)
```

Comment:
What is that "magic" number `maxsize=10000`?

Reply:
Added constant